### DISCUSSION TITLE
How to achieve user isolation in a chatflow?

### DISCUSSION THREAD

**Discussion by ops-004** [langflow-ai/langflow#1133] [Q&A]
Hello! I was impressed when I first used this tool. I hope to use it to build a chatflow and provide it for internal use within my company. However, I encountered an issue: I need to isolate the state of each user, such as managing their session history in the memory component. It is crucial that each user's session history remains isolated, as multiple users using it simultaneously may affect each other. Could you please advise me on how to solve this issue in such a scenario?

**Comment by ogabrielluiz** [comment]
Hi! 
I'm glad you liked it.

Memory components have a ChatMessageHistory field that you could connect to a PostgresChatMessageHistory or even use a MotorheadMemory. 

This will save the conversation to a session_id you provide so any time a new session is created, you can update the session_id and it should start from a blank chat history.

**Comment by Cristhianzl** [comment]
Hi! 👋

We are using other channels as our official means of communication with users. We apologize for the delayed response.
For faster assistance, please direct your questions to our Discord channel or open a GitHub issue, as we are much more active on these platforms.

[Langflow Discord Server](https://discord.com/invite/EqksyE2EX9)

Thank you for your understanding.

Best regards,
The Langflow Team
